wurst/network/SyncSimple.wurst

```diff
@@ -0,0 +1,177 @@
+		see Network. 
+
+		It depends on the fact that all network events in WC3 are fired sequentially,
+		meaning that they arrrive for other players in the order that they were
+		sent in.
+		It also depends on the fact that the EVENT_PLAYER_UNIT_SELECTED fires
+		synchronously for all players at the same time, allowing us to know for
+		certain when other players have acknowledged our unit selection, as well
+		as all network events that have been fired before it.
+
+		By calling the .sync() method, we queue a network action (specifically,
+		a unit selection event) that will only be delivered after all previously 
+		queued network actions have also been delivered.
+		The primary use of this library is in conjunction with gamecache's Sync
+		natives, because they also fire sequential network events. When we call
+		.sync() after a series of Sync natives, we ensure that the .onSynced()
+		callback will only be called after all players have received the data.
+
+		This way, we can send local data from one player to the rest, such as
+		camera position, data from files, and so on.
+
+		There may be other usages related to async network events as well.
+
+	Usage:
+		1. Create a new SimpleSynchronizer using:
+			let synchronizer = new SimpleSynchronizer()
+
+		2. Send some network events on one (or more) player using the Sync natives (or something else)
+			if sender == localPlayer
+				// value1 and value2 are some local values only known to localPlayer
+				SyncStoredInteger(gc, mkey, vkey1, value1)
+				SyncStoredInteger(gc, mkey, vkey1, value1)
+		
+		3. Set a callback to be called after all players have synchronized and read the data
+			synchronizer.onSynced(() -> begin
+				let value1 = LoadInteger(gc, mkey, vkey1)
+				let value2 = LoadInteger(gc, mkey, vkey2)
+
+				// here, value1 and value2 are integers received from sender
+				...
+			end)
+
+		4. Call .sync() on all game clients, after having called the Sync natives.
+			synchronizer.sync()
+
+		5. A SimpleSynchronizer can only be used once, and gets destroyed automatically after all players have synced.
+		To synchronize again, create a new SimpleSynchronizer
+			
+********************************************************************************/
+
+// briefly selects the specified unit by the player
+function player.onceSelect(unit what)
+	ENUM_GROUP.enumUnitsSelected(this, null)
+	unit last = null
+	var count = 0
+	for selected in ENUM_GROUP
+		count++
+		last = selected
+
+	if count >= 12
+		this.unselect(last)
+
+	if localPlayer == this
+		this.select(what)
+		this.unselect(what)
+
+	if count >= 12
+		this.select(last)
+
+	ENUM_GROUP.destr()
+	
+public interface SynchronizationCallback
+	function onSynchronized()
+
+public class SimpleSynchronizer
+	use LinkedListModule
+
+	private static constant DUMMY_PLAYER = players[PLAYER_NEUTRAL_PASSIVE]
+	private static constant DUMMY_ID     = 'hfoo'
+	private static constant DUMMY_GHOST  = 'Aeth'
```

Comment:
Already exists in Basics.wurst as GHOST_VIS_ID

Reply:
Done